src/main/java/com/exactpro/th2/check1/Check1Handler.java

```diff
@@ -167,11 +184,44 @@ public void submitNoMessageCheck(NoMessageCheckRequest request, StreamObserver<N
             if (logger.isErrorEnabled()) {
                 logger.error("No message check rule failed. Request " + MessageUtils.toJson(request), e);
             }
+            resultsStorage.removeResult(ruleId);
             responseObserver.onNext(NoMessageCheckResponse.newBuilder()
                     .setStatus(RequestStatus.newBuilder().setStatus(ERROR).setMessage("No message check rule failed. See the logs.").build())
                     .build());
         } finally {
             responseObserver.onCompleted();
         }
     }
+
+    @Override
+    public void waitForResult(WaitForResultRequest request, StreamObserver<WaitForResultResponse> responseObserver) {
+        try {
+            var responseBuilder = WaitForResultResponse.newBuilder();
+            var requestStatusBuilder = RequestStatus.newBuilder();
+
+            try {
+                var ruleResult = resultsStorage.getResult(request.getRuleId(), request.getTimeout());
+                if (ruleResult != null) {
+                    responseBuilder.setRuleResult(ruleResult);
+                    requestStatusBuilder.setStatus(SUCCESS);
+                } else {
+                    requestStatusBuilder.setStatus(ERROR).setMessage("No rule with specified id found");
+                }
+            } catch (TimeoutException e) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("WaitForResult timeout expired");
+                }
```

Comment:
No needs to wrap logging into `if` statement if we use constant string or use logger formatting. The formatting will be evaluated only if logging level is enabled